backend/routes/quests.go

```diff
@@ -211,6 +179,68 @@ func GetMainUserQuests(w http.ResponseWriter, r *http.Request) {
 	routeutils.WriteDataJson(w, string(quests))
 }
 
+func GetMainUserQuestsProgress(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	_, err := core.PostgresQueryJson[UserMainQuest]("SELECT questId FROM UserMainQuests where user_address = $1", userAddress)
+
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusInternalServerError, "Failed to get main user quests")
+		return
+	}
+
+	userMainQuestsProgress := []UserMainQuestProgress{
+		{QuestId: 1, Progress: 50, Needed: 50},
+		{QuestId: 2, Progress: 30, Needed: 70},
+	}
+
+	questsProgressJson, err := json.Marshal(userMainQuestsProgress)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusInternalServerError, "Failed to marshal quests progress")
+		return
+	}
+
+	routeutils.WriteDataJson(w, string(questsProgressJson))
+}
+
+func GetDailyUserQuestsProgress(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	dayIndex := r.URL.Query().Get("dayIndex")
+	if dayIndex == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing day index parameter")
+		return
+	}
+
+	_, err := core.PostgresQueryJson[UserDailyQuest]("SELECT questId FROM UserDailyQuests where user_address = $1 AND dayIndex = $2 ", userAddress, dayIndex)
```

Comment:
you'll want to query from `DailyQuests` here to get the quest id and type

Reply:
this mean i have to loop over the UserDailyQuest array and map each row to pull out the questId, is this not an expensive operation since am only returning a handcoded values for Progress and Needed

Reply:
Not sure I understand what you mean, but you shouldn't need to query `UserDailyQuests` at all. Only `DailyQuests` for the id and type fields

Reply:
so i should query DailyQuests with day_index where clause

Reply:
Yeah that would probably be best.
You could implement another route that takes the day index as an argument if you'd like too, but not essential now

Reply:
the daily quest progress endpoint take both user address and dayIndex as describe in the issue: #176